### PR TITLE
Secure code

### PR DESCRIPTION
**Pull request by atodd-noms**
1. Use secure cookie when node env == prod
2. Update dependencies to address vulnerabilities reported by Snyk
3. Add snyk protect to npm scripts
4. Add Helmet for HTTP header security vulnerability protection 
5. Disable x-powered-by header

As per https://expressjs.com/en/advanced/best-practice-security.html

? Should we also be setting cookie domain in prod ?

### CODE REVIEW

File: server/config.js [hunk 1]
```diff
@@ -20,8 +20,8 @@ module.exports = {
         database: get('DB_NAME', 'iis-sandbox')
     },
 
-    https: get('HTTPS', 'false'),
-    
+    https: process.env.NODE_ENV === 'prod' ? true : false,
```
Comment (glenjamin) [inline server/config.js]: the string is `"production"` for prod, there's a variable at the top of the file you can use

File: server/app.js [hunk 2]
```diff
@@ -62,9 +68,16 @@ let ssoConfig = config.sso;
...
     name: 'session',
     keys: [config.sessionSecret],
-    maxAge: 60 * 60 * 1000 // 60 minute
+    maxAge: 60 * 60 * 1000,
+    secure: config.https,
+    httpOnly: true,
+    signed: true,
+    overwrite: true,
+    sameSite: true
+    // domain: 'hpa.service.hmpps.dsd.io'
```
Comment (glenjamin) [inline server/app.js]: I don't think we need this, implicit domain is fine.

File: server/app.js [hunk 3]
```diff
@@ -30,12 +32,16 @@ let app = express();
...
 // http://stackoverflow.com/a/18455265/173062
 app.use(function(req, res, next) {
-    if(req.headers['x-arr-ssl'] && !req.headers['x-forwarded-proto']) {
+    if (req.headers['x-arr-ssl'] && !req.headers['x-forwarded-proto']) {
         req.headers['x-forwarded-proto'] = 'https';
     }
     return next();
 });
 
+// Secure code best practice
```
Comment (glenjamin) [inline server/app.js]: perhaps include the URL to the guide you used in this comment?

File: server/app.js [hunk 4]
```diff
@@ -30,12 +32,16 @@ let app = express();
...
+app.disable('x-powered-by');
+app.use(helmet());
```
Comment (glenjamin) [inline server/app.js]: I think we should enable the `noCache` option in helmet as well.

File: package.json [hunk 5]
```diff
@@ -10,7 +10,9 @@
   "scripts": {
     "postinstall": "gulp",
     "start": "gulp && node .",
-    "test": "gulp build test"
+    "test": "gulp build test",
+    "snyk-protect": "snyk protect",
```
Comment (glenjamin) [inline package.json]: Does this step get run on CI?

File: server/config.js (hunk 1 above)
Comment (atodd-noms) [inline server/config.js]: Done

File: server/app.js (hunk 3 above)
Comment (atodd-noms) [inline server/app.js]: Done

**Pull request merged**